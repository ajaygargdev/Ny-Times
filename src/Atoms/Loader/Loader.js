import React from "react";
import { Typography, Box, CircularProgress } from "@mui/material";

const Loader = () => (
  <Box
    display="flex"
    justifyContent="center"
    alignItems="center"
    height="100vh"
  >
    <CircularProgress />
    <Typography variant="h6" sx={{ ml: 2 }}>
      Loading articles...
    </Typography>
  </Box>
);

export default Loader;
