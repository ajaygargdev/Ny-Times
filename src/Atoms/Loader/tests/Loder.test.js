/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import Loader from "../Loader";

describe("Loader Component", () => {
  test("renders without crashing", () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });
});
