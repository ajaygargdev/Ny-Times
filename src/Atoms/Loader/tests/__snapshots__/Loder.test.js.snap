// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders without crashing 1`] = `
<div>
  <div
    class="MuiBox-root css-1lb6n3v"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-ib86xl-MuiTypography-root"
    >
      Loading articles...
    </h6>
  </div>
</div>
`;
